### PR TITLE
SITES-16562 - [Xwalk] Open Universal Editor from Franklin Sidekick

### PR DESCRIPTION
**Pull request by jckautzmann**
- adds a customized Edit button to the Sidekick to open the page with the Universal Editor
- hides the default buttons of the Sidekick

Fix [SITES-16562](https://jira.corp.adobe.com/browse/SITES-16562)

Test URLs:
- Before: https://main--aem-boilerplate-xwalk--jckautzmann.hlx.page/
- After: https://open-ue--aem-boilerplate-xwalk--jckautzmann.hlx.page/

Example:
Screenshot 2024-04-17 at 18 02 28

### CODE REVIEW

**Comment by jckautzmann** [comment]
The `aem-psi-check` fails with the message:
```
Pull request description must include at least one preview url to the test project.
Please update the description.
```
although the description contains test URLs.

File: scripts/sidekick.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    };
+    check();
+  });
+}
+
+async function customizeButtons(sk) {
+  // hide the default buttons
+  const container = await getElement(sk, '.plugin-container');
+  container.style.visibility = 'hidden';
+  for (let i = 0; i < ['edit', 'reload', 'publish', 'delete', 'unpublish'].length; i += 1) {
```
Comment (Buuhuu) [inline scripts/sidekick.js:45]: use `['edit', 'reload', 'publish', 'delete', 'unpublish'].forEach(async (action) => { ... })` instead

File: scripts/scripts.js [hunk 2]
```diff
@@ -142,6 +143,7 @@ function loadDelayed() {
   // eslint-disable-next-line import/no-cycle
   window.setTimeout(() => import('./delayed.js'), 3000);
   // load anything that can be postponed to the latest here
+  initSidekick();
```
Comment (Buuhuu) [inline scripts/scripts.js:146]: better do an `import('./sidekick.js').then(({ initSidekick }) => initSidekick());` here

that defers loading the script until after loadLazy finished. with the static import at the top scripts.js only executes with the script loaded and hence we impact LCP.

File: scripts/sidekick.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+async function openWithUniversalEditor(event) {
+  const { owner, repo, ref } = event.detail.data.config;
+  const contentSourceUrl = await getContentSourceUrl(owner, repo, ref);
+  if (!contentSourceUrl) {
+    // eslint-disable-next-line no-console
+    console.error('Content source URL not found');
+    return;
+  }
+  const path = window.location.pathname;
```
Comment (Buuhuu) [inline scripts/sidekick.js:17]: use destructuring

```
const { pathname } = window.location;
```

File: scripts/sidekick.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  const editorUrl = `${contentSourceUrl}${path}?cmd=open`;
+  // open the editor in a new tab
+  window.open(editorUrl, '_blank');
+}
+
+function getElement(sk, selector) {
```
Comment (Buuhuu) [inline scripts/sidekick.js:23]: this should be an `async` function as it may return a promise.

File: scripts/sidekick.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  let elt = sk.shadowRoot.querySelector(selector);
+  if (elt) {
+    return elt;
```
Comment (Buuhuu) [inline scripts/sidekick.js:26]: you can remove the early return here. it is effectively the same as the promise code below.

File: scripts/sidekick.js (hunk 1 above)
Comment (jckautzmann) [reply scripts/sidekick.js:45]: This was giving a linting error. That's why I used the `for` loop.

File: scripts/scripts.js (hunk 2 above)
Comment (jckautzmann) [reply scripts/scripts.js]: addressed in https://github.com/adobe-rnd/aem-boilerplate-xwalk/pull/17/commits/8fd5333b8d0d2074b0cad89dc3a3184be885f3e3

File: scripts/sidekick.js (hunk 4 above)
Comment (jckautzmann) [reply scripts/sidekick.js]: addressed in https://github.com/adobe-rnd/aem-boilerplate-xwalk/pull/17/commits/8fd5333b8d0d2074b0cad89dc3a3184be885f3e3

File: scripts/sidekick.js (hunk 5 above)
Comment (jckautzmann) [reply scripts/sidekick.js]: addressed in https://github.com/adobe-rnd/aem-boilerplate-xwalk/pull/17/commits/8fd5333b8d0d2074b0cad89dc3a3184be885f3e3

File: scripts/sidekick.js (hunk 1 above)
Comment (Buuhuu) [reply scripts/sidekick.js:42]: but then at least define array as a variable and don't duplicate it.

Comment (jckautzmann) [reply scripts/sidekick.js]: addressed in https://github.com/adobe-rnd/aem-boilerplate-xwalk/pull/17/commits/5c102d6d6d559aa20bb802fe9cf4a440707944e6

**Pull request merged**